frappe/model/document.py

```diff
@@ -198,7 +198,7 @@ def raise_no_permission_to(self, perm_type):
 		raise frappe.PermissionError
 
 	def insert(self, ignore_permissions=None, ignore_links=None, ignore_if_duplicate=False,
-				ignore_mandatory=None, set_name=None, set_child_names=True):
+				ignore_mandatory=None, set_name=None, set_child_names=True, db_auto_commit = False):
```

Comment:
Bad idea to commit mid Document insert.

Reply:
Absolutely - however it is the only way to get some tests to function properly and this was the lowest touch way I could manage to do it. If there is a better way I am 100% open to it.

Reply:
This is now reverted to its original state - no more autocommit